examples/server/server.cpp

```diff
@@ -2592,10 +2592,15 @@ int main(int argc, char ** argv) {
         return false;
     };
 
-    auto middleware_server_state = [&res_error, &state](const httplib::Request &, httplib::Response & res) {
+    auto middleware_server_state = [&res_error, &state](const httplib::Request & req, httplib::Response & res) {
         server_state current_state = state.load();
         if (current_state == SERVER_STATE_LOADING_MODEL) {
-            res_error(res, format_error_response("Loading model", ERROR_TYPE_UNAVAILABLE));
+            if(req.path == "/"){
```

Comment:
You're missing one requirement from the original issue #9158 

Here only `/` is catched, not `*.html`. I leave a suggestion on my comment https://github.com/ggerganov/llama.cpp/pull/9401#discussion_r1751417948 , but I'll apply it myself.